imports/plugins/core/orders/client/containers/orderDashboardContainer.js

```diff
@@ -194,6 +260,397 @@ class OrderDashboardContainer extends Component {
+        isNotLabeled++;
+        whichFalseState = shippingStates.picked;
+      } else if (order.shipping[0].workflow.status === "coreOrderWorkflow/picked") {
+        isNotLabeled++;
+        whichFalseState = shippingStates.packed;
+      } else if (order.shipping[0].workflow.status === "coreOrderWorkflow/packed") {
+        isNotLabeled++;
+        whichFalseState = shippingStates.labeled;
+      } else if (order.shipping[0].workflow.status === "coreOrderWorkflow/labeled") {
+        isNotShipped++;
+      } else if (order.shipping[0].workflow.status === "coreOrderWorkflow/shipped") {
+        isShipped++;
+      }
+    });
+
+    // display proper alert if the order(s) are in this state already or want to skip the previous states
+    this.displayAlert(selectedOrders, status, whichFalseState, isNotLabeled, isNotShipped, isShipped);
+  }
+
+  /**
+   * setShippingStatus
+   *
+   * @summary call the relevant method based on the provided shipping status
+   * @param {String} status - the selected shipping status to be set
+   * @param {Array} selectedOrdersIds - array of ids of the selected orders
+   * @return {null} no return value
+   */
+  setShippingStatus = (status, selectedOrdersIds) => {
+    this.setState({
+      renderFlowList: true
+    });
+    const selectedOrders = this.state.orders.filter((order) => {
+      return selectedOrdersIds.includes(order._id);
+    });
+
+    if (status === "picked") {
+      this.pickedShippingStatus(selectedOrders, status);
+    }
+
+    if (status === "packed") {
+      this.packedShippingStatus(selectedOrders, status);
+    }
+
+    if (status === "labeled") {
+      this.labeledShippingStatus(selectedOrders, status);
+    }
+
+    if (status === "shipped") {
+      this.shippedShippingStatus(selectedOrders, status);
+    }
+  }
+
+  handleBulkPaymentCapture = (selectedOrdersIds) => {
+    this.setState({
+      isLoading: {
+        capturePayment: true
+      }
+    });
+    const selectedOrders = this.state.orders.filter((order) => {
+      return selectedOrdersIds.includes(order._id);
+    });
+
+    let orderCount = 0;
+
+    selectedOrders.forEach((order) => {
+      Meteor.call("orders/approvePayment", order, (err) => {
+        if (err) {
+          this.setState({
+            isLoading: {
+              capturePayment: false
+            }
+          });
+          Alerts.toast(`An error occured while approving the payment: ${err}`, "error");
+        } else {
+          Meteor.call("orders/capturePayments", order._id, (error) => {
+            if (error) {
+              this.setState({
+                isLoading: {
+                  capturePayment: false
+                }
+              });
+              Alerts.toast(`An error occured while capturing the payment: ${error}`, "error");
+            }
+
+            if (order.workflow.status === "new") {
+              Meteor.call("workflow/pushOrderWorkflow", "coreOrderWorkflow", "processing", order);
```

Comment:
This should probably be performed on the server and not as an additional call from the client.